File: ROS_Projects/src/my_package/scripts/subscriber.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def callbackFn(msg):
    rospy.loginfo('Message received: ' + str(msg))

if __name__ == '__main__':
    rospy.init_node('subscriber')
    sub = rospy.Subscriber('/random_topic',String, callbackFn)
    rospy.spin()

File: ROS_Projects/src/my_package/src/subscriber.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>

void callBackFun(const std_msgs::String &msg)
{
    ROS_INFO("Received message: %s", msg.data.c_str());
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "subscriber");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("/random", 10, callBackFun);
    ros::spin();
}